sre-recipes/recipes/currencies_recipe/currencies_recipe.py

```diff
@@ -48,103 +41,74 @@ def _deploy_state(state):
         logging.info('Setting env variable: %s', set_env_command)
         logging.info('Getting pod: %s', get_pod_command)
 
-        CurrenciesRecipe._run_command(set_env_command)
-        service, error = CurrenciesRecipe._run_command(get_pod_command)
+        Recipe._run_command(set_env_command)
+        service, error = Recipe._run_command(get_pod_command)
         service = service.decode("utf-8").replace('"', '')
         delete_pod_command = f"kubectl delete pod {service}"
         logging.info('Deleting pod: %s', delete_pod_command)
-        CurrenciesRecipe._run_command(delete_pod_command)
-
-    @staticmethod
-    def _auth_cluster():
-        """ Authenticates for kubectl commands. """
-        logging.info('Authenticating cluster')
-        get_project_command = "gcloud config list --format value(core.project)"
-        project_id, error = CurrenciesRecipe._run_command(get_project_command)
-        project_id = project_id.decode("utf-8").replace('"', '')
-        zone_command = "gcloud container clusters list --filter name:cloud-ops-sandbox --project {} --format value(zone)".format(project_id)
-        zone, error = CurrenciesRecipe._run_command(zone_command)
-        zone = zone.decode("utf-8").replace('"', '')
-        auth_command = "gcloud container clusters get-credentials cloud-ops-sandbox --project {} --zone {}".format(project_id, zone)
-        CurrenciesRecipe._run_command(auth_command)
-        logging.info('Cluster has been authenticated')
+        Recipe._run_command(delete_pod_command)
 
     def break_service(self):
         """
         Rolls back the working version of the given service and deploys the
         broken version of the given service
         """
-        print("Deploying broken service...")
-        self._auth_cluster()
+        print('Deploying broken service...')
+        Recipe._auth_cluster()
         self._deploy_state(True)
-        print("Done")
+        print('Done')
         logging.info('Deployed broken service')
 
     def restore_service(self):
         """
         Rolls back the broken version of the given service and deploys the
         working version of the given service
         """
-        print("Deploying working service...")
-        self._auth_cluster()
+        print('Deploying working service...')
+        Recipe._auth_cluster()
         self._deploy_state(False)
-        print("Done")
+        print('Done')
         logging.info('Deployed working service')
 
-    def _service_multiple_choice(self):
+    def hint(self):
+        """
+        Provides a hint about finding the root cause of this recipe
+        """
+        print('Giving hint for recipe')
+        external_ip_command = "kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}'"
```

Comment:
I tried running `sandboxctl hint recipe0` without authenticating to a cluster, and it seemed to hang forever. Can we error handle this somehow? Or maybe authenticate to the cluster by default?

Reply:
Yeah I added an auth command to this